### DISCUSSION TITLE
Microtonal accidentals

### DISCUSSION THREAD

**Discussion by topchyan** [Audiveris/audiveris#659] [Ideas]
Would it be possible to add various microtonal accidentals to the list of recognized glyphs? Or how to go about recognizing them in the existing version?

![image](https://github.com/Audiveris/audiveris/assets/16518288/9747c26b-5a1a-4f26-add3-b9fc706fa40f)

**Comment by hbitteur** [comment]
@topchyan 
Yes it is possible to _recognize_ additional shapes, it's just a matter of listing them and training the neural network on enough representative samples.

But to go beyond that, we need to handle these new beasts:
- Do they behave like "standard" accidentals, with just smaller pitch fractions?
- What about key signatures: can we find these microtonal signs in a key signature?
- How do we export them in MusicXML output?

Sorry, I'm totally ignorant about microtonal signs.

Anyway, I know that supporting them would imply modifications in the current code.
This has to be balanced with the number of users potentially interested by this new feature.
Don't forget there are still many features waiting for being implemented...

**Reply by topchyan** [reply]
Forgot to mention: in addition to what MusicXML standard [recommends](https://www.w3.org/2021/06/musicxml40/musicxml-reference/examples/alter-element-microtones/). I saw markup like <accidental>**sharp-down**</accidental> which would show a sharp sign with an arrow pointing down, for example, but that notation isn't fully recognized by convertors, so I had to be creative and pretend it's a known symbol, like sharp-sharp and then after conversion adjust it back. It's tedious, but it works for the most part.

**Comment by topchyan** [comment]
@hbitteur thank you for the follow up on this. 

There are several microtonal notation systems out there and Audiveris is unlikely to support all of them. However, I thought perhaps the most common set of quarter tones would be a good start? Here's a sample of the score you can look at to answer your questions:

[Kemani Tatyos Efendi Karcigar Pesrev](
https://neyzen.com/nota_arsivi/02_klasik_eserler/050_karcigar/karcigar_p_kemani_tatyos.pdf
)

As you can see they do behave more or less as standard accidentals.

Also found this image, which seems helpful:

![image](https://github.com/Audiveris/audiveris/assets/16518288/7336eed4-7bb1-4b47-a4e5-df10d3dbb7aa)

As far as how much of this is going to be useful is a great question. I know there were in the past efforts made to digitize a large number of the pieces (https://compmusic.upf.edu/node/140). So having this feature may be helpful to some out there who study non-western modal music.
